quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliTlsCommandIT.java

```diff
@@ -0,0 +1,141 @@
+
+    @Inject
+    static QuarkusTlsCommand tlsCommand;
+
+    @Order(1)
+    @Test
+    public void generateQuarkusCa() {
+        // also prepares state for assertion in TlsCommandTest
+        deleteFileIfExists(DEV_CA_CERT_FILE);
+        deleteFileIfExists(DEV_CA_PK_FILE);
+        tlsCommand
+                .generateQuarkusCa()
+                .withOption(GenerateQuarkusCaOptions.TRUSTSTORE_LONG)
+                .withOption(GenerateQuarkusCaOptions.RENEW_SHORT)
+                .executeCommand()
+                .assertCommandOutputContains("Root CA certificate generated successfully")
+                .assertCommandOutputContains("Quarkus Development CA generated and installed")
+                .assertFileExistsStr(cmd -> cmd.getOutputLineRemainder("Truststore generated successfully:"));
+        assertTrue(DEV_CA_CERT_FILE.exists(),
+                "Quarkus CLI subcommand 'tls generate-quarkus-ca' didn't generate Quarkus DEV CA certificate");
+        assertTrue(DEV_CA_PK_FILE.exists(),
+                "Quarkus CLI subcommand 'tls generate-quarkus-ca' didn't generate Quarkus DEV CA private key");
+    }
+
+    @Order(2)
+    @Test
+    public void generateCertificate() {
+        // also prepares state for assertion in TlsCommandTest
+        var appSvcDir = tlsCommand.getApp().getServiceFolder().toAbsolutePath().toString();
+        tlsCommand
+                .generateCertificate()
+                .withOption(GenerateCertOptions.COMMON_NAME_LONG, CN)
+                .withOption(GenerateCertOptions.NAME_SHORT, CERT_NAME)
+                .withOption(GenerateCertOptions.PASSWORD_SHORT, PASSWORD)
+                .withOption(GenerateCertOptions.DIRECTORY_LONG, appSvcDir)
+                .executeCommand()
+                .assertCommandOutputContains("Quarkus Dev CA certificate found at " + DEV_CA_CERT_FILE.getAbsolutePath())
+                .assertCommandOutputContains("PKCS12 keystore and truststore generated successfully!")
+                .assertFileExistsStr(cmd -> cmd.getOutputLineRemainder("Key Store File:"))
+                .assertFileExistsStr(cmd -> cmd.getOutputLineRemainder("Trust Store File:"))
+                // save truststore path in application properties so that we can use it in TlsCommandTest
+                .addToAppProps(cmd -> TRUST_STORE_PATH + "=" + cmd.getOutputLineRemainder("Trust Store File:"))
+                .assertCommandOutputContains(
+                        "Signed Certificate generated successfully and exported into `%s-keystore.p12`".formatted(CERT_NAME))
+                // following properties are set by this tls command, and we want to use them TlsCommandTest as well
+                .addToAppProps(getPropertyFromEnvFileAndChangeProfileToTest("quarkus.tls.key-store.p12.path"))
+                .addToAppProps(getPropertyFromEnvFileAndChangeProfileToTest("quarkus.tls.key-store.p12.password"));
+    }
+
+    @Order(3)
```

Comment:
Does this test need to have Order? AFAIK testing help in stateless and should not interfere with other tests.

Reply:
yes, this is stateful test. basically I am testing individual commands and in the end, I use certificates generated by individual command to test they work. so:

- I need CA to test generation of certs because I don't want to test self-signed certs (well, DEV CA seems more practical)
- I need generated TS/KS before I start application that is using them

I will add comments that explains it. You are right it is not clear at first sight.

Reply:
I'm not sure if we understand each other. I know you need to first generate CA, then Cert then build app.
But this single test "testHelpOption" IMHO seems a bit out of place - in middle of generating and running cert+app. From the code, it seems the "testHelpOption" actually doesn't work with generated CA or cert.

Reply:
yes, I added `testHelpOption` absolutely randomly after I created PR and remember it should be there. You are right, I'll drop the order. Thank you